#### About

> Critical Value.

The critical value is a value of a test statistic that is used to determine whether to reject or fail to reject the null hypothesis in a hypothesis test. It is typically based on the significance level of the test and the degrees of freedom.



In [1]:
import scipy.stats as stats


In [2]:
sample_mean = 8.6
pop_mean = 8.0
sample_std = 1.2
n = 20


In [3]:
t_value = (sample_mean - pop_mean) / (sample_std / (n**0.5))
print("t-value:", t_value)

t-value: 2.236067977499789


In [4]:
# calculate critical value
alpha = 0.05
df = n - 1
critical_value = stats.t.ppf(alpha/2, df)
print("critical value:", critical_value)

critical value: -2.0930240544082634


In [5]:
# compare t-value and critical value
if abs(t_value) > critical_value:
    print("Reject null hypothesis")
else:
    print("Fail to reject null hypothesis")

Reject null hypothesis


In this example, the calculated t-value is 2.23, and the critical value is approximately -2.093. Since the t-value is greater than the critical value, we can reject the null hypothesis at the 0.05 significance level and conclude that the sample mean is significantly different from the hypothesized population mean.

